lib/executor/ramping_vus.go

```diff
@@ -636,40 +633,88 @@ func (vlv RampingVUs) Run(ctx context.Context, out chan<- stats.SampleContainer)
 		currentMaxAllowedVUs = newMaxAllowedVUs
 	}
 
-	handleAllRawSteps := func() bool {
-		for {
+	// This is the step that is the actual gracefulStop at the end
+	lastStep := gracefulExecutionSteps[len(gracefulExecutionSteps)-1]
+	// TODO maybe instead of getting only the last step (above) mergeSteps should return the
```

Comment:
Yeah, I think `mergeSteps()` should return the remaining `gracefulExecutionSteps`. Saving only the last one is not good enough, since all of the VUs could have been gracefully stopped and finished their iterations 30s (or more, depending on `gracefulStop`) before its time...

Reply:
as far as I am concerned the last one is the one that is the actual gracefulStop ... and the one that needs to be run in a separate goroutine in order for the gracefulStop to be executed. 
I can't imagine a case where the others will matter ... maybe I am wrong, but if that is the case I would also like to add a test for that

Reply:
This is an issue with the naming of these steps... When there is a ramping down in the "raw" steps, the `gracefulStop` of the respective VU handles is triggered. When there is a raping down in the `gracefulExecutionSteps`, the `hardStop()` is triggered, right? 

So, if you have short iterations, the executor can finish very soon after the last "raw" step, while there could be multiple "graceful" steps waiting 30s (or more) in the future

Reply:
can you come up with an example of this ?

Reply:
Yeah, try this with the code from your branch and with the code from `new-schedulers`:

```js
import { sleep } from 'k6';
export let options = {
    stages: [
        { target: 100, duration: "5s" },
    ],
};

export default function () {
    sleep(1);
};
```

Reply:
or maybe not the code from `new-schedulers`, since it also has a bug :sweat_smile: but if you run it with your branch, you'll see what I mean

Reply:
fixed in 1c66584